specification/azurefleet/AzureFleet.Management/baseVirtualMachineProfile.tsp

```diff
@@ -0,0 +1,2213 @@
+   */
+  bootDiagnostics?: BootDiagnostics;
+}
+
+/**
+ * Boot Diagnostics is a debugging feature which allows you to view Console Output
+ * and Screenshot to diagnose VM status. You can easily view the output of your
+ * console log. Azure also enables you to see a screenshot of the VM from the
+ * hypervisor.
+ */
+model BootDiagnostics {
+  /**
+   * Whether boot diagnostics should be enabled on the Virtual Machine.
+   */
+  enabled?: boolean;
+
+  /**
+   * Uri of the storage account to use for placing the console output and
+   * screenshot. If storageUri is not specified while enabling boot diagnostics,
+   * managed storage will be used.
+   */
+  storageUri?: string;
+}
+
+/**
+ * Describes a virtual machine scale set extension profile.
+ */
+model VirtualMachineScaleSetExtensionProfile {
+  /**
+   * The virtual machine scale set child extension resources.
+   */
+  extensions?: VirtualMachineScaleSetExtension[];
+
+  /**
+   * Specifies the time alloted for all extensions to start. The time duration
+   * should be between 15 minutes and 120 minutes (inclusive) and should be
+   * specified in ISO 8601 format. The default value is 90 minutes (PT1H30M).
+   * Minimum api-version: 2020-06-01.
+   */
+  extensionsTimeBudget?: string;
+}
+
+/**
+ * Describes a Virtual Machine Scale Set Extension.
+ */
+model VirtualMachineScaleSetExtension is SubResourceReadOnly {
+  /**
+   * The name of the extension.
+   */
+  name?: string;
+
+  /**
+   * Resource type
+   */
+  @visibility("read")
+  type?: string;
+
+  /**
+   * Describes the properties of a Virtual Machine Scale Set Extension.
+   */
+  properties?: VirtualMachineScaleSetExtensionProperties;
+}
+
+/**
+ * Describes the properties of a Virtual Machine Scale Set Extension.
+ */
+model VirtualMachineScaleSetExtensionProperties {
+  /**
+   * If a value is provided and is different from the previous value, the extension
+   * handler will be forced to update even if the extension configuration has not
+   * changed.
+   */
+  forceUpdateTag?: string;
+
+  /**
+   * The name of the extension handler publisher.
+   */
+  publisher?: string;
+
+  /**
+   * Specifies the type of the extension; an example is "CustomScriptExtension".
+   */
+  type?: string;
+
+  /**
+   * Specifies the version of the script handler.
+   */
+  typeHandlerVersion?: string;
+
+  /**
+   * Indicates whether the extension should use a newer minor version if one is
+   * available at deployment time. Once deployed, however, the extension will not
+   * upgrade minor versions unless redeployed, even with this property set to true.
+   */
+  autoUpgradeMinorVersion?: boolean;
+
+  /**
+   * Indicates whether the extension should be automatically upgraded by the
+   * platform if there is a newer version of the extension available.
+   */
+  enableAutomaticUpgrade?: boolean;
+
+  /**
+   * Json formatted public settings for the extension.
+   */
+  settings?: Record<unknown>;
```

Comment:
Can I use this and suppress its warnings? or need to replace it with something else?

Reply:
@tadelesh I remember we decided on how to express swagger `type:object` in typespec? Could you help there?
https://github.com/Azure/azure-rest-api-specs/blob/d715112415eb08572ee96cc4bd77217aa44dedf1/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json#L5002C1-L5009C11

Reply:
Yes this will always be a dictionary. 

I checked in Compute and they are defined as empty objects:
[azure-rest-api-specs/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json at d715112415eb08572ee96cc4bd77217aa44dedf1 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/blob/d715112415eb08572ee96cc4bd77217aa44dedf1/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2024-03-01/virtualMachineScaleSet.json#L5002C1-L5009C11)

Reply:
Is there a way to define object in typespec but with no additionalProperties added in generated swagger?  @tadelesh 
![image](https://github.com/Azure/azure-rest-api-specs/assets/7030203/7cb366c2-3688-4c24-97c1-e8db5e495dbc)